### DISCUSSION TITLE
any reason why EqualsAvoidNull not implemented for enums?

### DISCUSSION THREAD

**Discussion by uri-ravzin** [checkstyle/checkstyle#14789] [Ideas]
it works great on strings, and could be very beneficial...

**Comment by rnveach** [comment]
> works great on strings

It actually works only on strings the check can identify. If there is doubts if some variable/method is a string, it is ignored. I remember some conversations about this during a big change it was going through.

Also, Checkstyle is not currently a multi-file utility. It only knows what is in the current class it is looking at. It would be very hard to identify something as an enum and would be restricted to very limited cases. For example, we couldn't ensure all cap static fields are `enum`s since there are some that are just plain fields. Example, `java.awt.Color`.

Finally, I am of the belief you should not use `equals` with enums. They are immutable, so it is a waste to call the equals method, when you can just use `==`.

**Reply by nrmancuso** [reply]
>Finally, I am of the belief you should not use equals with enums. They are immutable, so it is a waste to call the equals method, when you can just use ==.

Yep, to add to this:

From https://docs.oracle.com/javase/specs/jls/se9/html/jls-8.html#jls-8.9:

> An enum type has no instances other than those defined by its enum constants.

`==` will provide a null safe way to check equality, and also gives you compile time type checking.
